### PR TITLE
Continued response to Tuple topic feedback

### LINKED ISSUE #1889: [Bug] Clarifications to What's New in C# 7 and Tuples topics
@richlander suggested several updates to the What's new in C# 7 topic [here](https://github.com/dotnet/docs/pull/1842#issuecomment-293363377)  That PR has a tighter focus (how to describe the dependency on .NET 4.7, or the `System.ValueTuple` package).

A summary of the changes requested are:

- [x] Move the dependency on `ValueTuple` and `ValueTask` to opening of the *Tuples* and *generalized async return types* respectively.
- [x] More clearly separate language updates from library types. (See Issue #1760 for more details here)
- [x] Tighten up some of the explanations (see referenced comment above)

### PR DESCRIPTION
**Pull request by BillWagner**
Fixes #1889

Much of these had been addressed by @svick in PR #1842. Adding the remaining open
items.

### CODE REVIEW

**Comment by BillWagner** [comment]
@terrajobst Can you review my edits on types on which language features depend and .NET standard? I want to make sure it's accurate.

/cc @richlander

File: docs/csharp/tuples.md [hunk 1]
```diff
@@ -25,10 +25,20 @@ In this topic, you'll learn the language rules governing Tuples in C# 7,
 different ways to use them, and initial guidance on working with Tuples.
 
 > [!NOTE]
-> The new tuples features require the `System.ValueTuple` type. For Visual Studio 2017,
-> you must add the NuGet package [System.ValueTuple](https://www.nuget.org/packages/System.ValueTuple/), available on the NuGet Gallery.
-> Without this package you may get a compilation error similar to `error CS8179: Predefined type 'System.ValueTuple``2' is not defined or imported`
```
Comment (rpetrusha) [inline docs/csharp/tuples.md]: We should add CS8137 and CS8179 to the sorry-we-don-t-have-specifics... file in the C# compiler error messages directory. I'll open a quick PR to do it and add a few others.

File: docs/csharp/tuples.md [hunk 2]
```diff
@@ -25,10 +25,20 @@ In this topic, you'll learn the language rules governing Tuples in C# 7,
...
-> or `error CS8137: Cannot define a class or member that utilizes tuples because the compiler required type 'System.Runtime.CompilerServices.TupleElementNamesAttribute' cannot be found.`
+> The new tuples features require the @System.ValueTuple types.
+> You must add the NuGet package [`System.ValueTuple`](https://www.nuget.org/packages/System.ValueTuple/) in order to use it
+> on platforms that do not include the types.
+>
+> This is similar to other language features that rely on types
+> delivered in the framework. Example include `async` and `await`
```
Comment (rpetrusha) [inline docs/csharp/tuples.md]: Example --> Examples

**Review by rpetrusha — APPROVED** [review]
Aside from one comment, LGTM.

File: docs/csharp/tuples.md [hunk 3]
```diff
@@ -25,10 +25,20 @@ In this topic, you'll learn the language rules governing Tuples in C# 7,
...
+> This is similar to other language features that rely on types
+> delivered in the framework. Examples include `async` and `await`
+> relying on the `INotifyCompletion` interface, and LINQ relying
+> on `IEnumerable<T>`. However, the delivery mechanism is changing
+> as .NET is becoming more platform independent. The .NET Framework
+> may not always ship on the same cadence as the language compiler. When new language
+> features rely on new types, those types will be available as NuGet packages when
+> the language features ship. As these new types get added to the .NET Standard
+> API and delivered as part of the framework, the NuGet package requirement will
+> be removed.
```
Comment (terrajobst) [inline docs/csharp/tuples.md]: Sounds good to me :-)

**Review by terrajobst — APPROVED** [review]

**Pull request merged**